src/main/scala/com/fulcrumgenomics/umi/CorrectUmis.scala

```diff
@@ -185,7 +200,7 @@ class CorrectUmis
           missingUmisRecords += 1
           rejectOut.foreach(w => w += rec)
         case Some(umi: String) =>
-          val sequences = umi.split("-", -1)
+          val sequences = if (revcomp) umi.split("-", -1).map(Sequences.revcomp) else umi.split("-", -1)
```

Comment:
here's where the new `--revcomp` argument is used

Reply:
I'd actually prefer your idea of reversing when there are multiple barcodes.  The client I'm implementing does not have (and will not have) dual UMIs, so I prefer the intuitive approach as it does not affect the client.  Good catch!